pipeline/metadata/blockpage.py

```diff
@@ -1,55 +1,48 @@
 """Matcher for response pages to blockpage signatures."""
 
 import json
-import os
+import io
+import pkgutil
 import re
 from typing import Optional, Dict
 
-import apache_beam.io.filesystems as apache_filesystems
-
 # Signature filenames
-FALSE_POSITIVES = 'false_positive_signatures.json'
-BLOCKPAGES = 'blockpage_signatures.json'
+FALSE_POSITIVES = 'data/false_positive_signatures.json'
+BLOCKPAGES = 'data/blockpage_signatures.json'
 
 
-def _load_signatures(path: str) -> Dict[str, re.Pattern]:
+def _load_signatures(filepath: str) -> Dict[str, re.Pattern]:
   """Load signatures for blockpage matching.
 
   Args:
-    path: path to json file containing signatures
+    filepath: relative path to json file containing signatures
 
   Returns:
     Dictionary mapping fingerprints to signature patterns
   """
+  data = pkgutil.get_data(__name__, filepath)
+  if not data:
+    raise FileNotFoundError(f"Couldn't find file {filepath}")
+  content = io.TextIOWrapper(io.BytesIO(data), encoding='utf-8')
+
   signatures = {}
-  with apache_filesystems.FileSystems.open(path) as f:
-    for line in f:
-      if line != b'\n':
-        signature = json.loads(line.strip())
-        # Patterns stored in BigQuery syntax,
-        # so % represents any number of characters
-        pattern = signature['pattern']
-        # Convert to Python regex
-        pattern = re.escape(pattern)
-        pattern = pattern.replace('%', '.*')
-        signatures[signature['fingerprint']] = re.compile(pattern, re.DOTALL)
+  for line in content.readlines():
+    if line != '\n':
+      signature = json.loads(line.strip())
+      pattern = signature['pattern']
+      fingerprint = signature['fingerprint']
+
+      signatures[fingerprint] = re.compile(pattern, re.DOTALL)
   return signatures
 
 
 class BlockpageMatcher:
   """Matcher to confirm blockpages or false positives."""
 
-  def __init__(self, signature_folder: str) -> None:
-    """Create a Blockpage Matcher.
-    Args:
-      signature_folder: a folder containing signature files.
-        Either a gcs filepath or a local system folder.
-    """
-    false_positive_path = os.path.join(signature_folder, FALSE_POSITIVES)
-    blockpage_path = os.path.join(signature_folder, BLOCKPAGES)
-
-    self.false_positives = _load_signatures(false_positive_path)
-    self.blockpages = _load_signatures(blockpage_path)
+  def __init__(self) -> None:
+    """Create a Blockpage Matcher."""
+    self.false_positives = _load_signatures(FALSE_POSITIVES)
+    self.blockpages = _load_signatures(BLOCKPAGES)
 
   def match_page(self, page: str) -> Optional[bool]:
```

Comment:
My thought was to use the existing interface for this PR and then update it after since there's already a lot changing here.

What are your thoughts on the right data to expose to users? My initial thought was to have only a `blockpage_signature` field and not expose the FP data since end-users don't care about the details of our FP detection. But we're using this internally too, and *we* care, since we want to be able to debug the system.

Maybe have the columns be
`blockpage`: `<pattern string>` which would be for end users/dashboards
`false_positive_blockpage`: `<pattern string>` which would be for internal debugging.

Reply:
I agree we should expose the signature id. Let's keep it as a TODO for now.

I would suggest `page_signature: <pattern_id>, is_block_page:<true|false|unknown>`
`page_signature` is the id, not the pattern.`is_block_page` can be derived from `page_signature`.

This will allow us to generate explanations like `HTTP Interference: domain $domain returns blockpage $page_signature` 

Also, in the future I can see use tagging pages even if they are not blockpages, like the cloudflare ddos protection page.

Reply:
By the way, having a way to tag generic pages (like Cloudflare) is another reason to have the simpler PageMatcher.

Reply:
The reason I was thinking of having `blockpage` / `false_positive_blockpage` was for the sake of our future dashboard use. If both blockpages and false positive signature are mixed up in the same column it will be difficult to display them separately, or to only display blockpages.

But I agree this can all be a TODO. It will be easier to reason about in a follow-up PR.

Reply:
The reason I proposed the `is_block_page` is for the dashboard or a data consumer. The page fingerprint would be mostly for inspection, validation, to see if things are consistent.

Reply:
added a TODO